app/scripts/metamask-controller.js

```diff
@@ -4788,6 +4788,24 @@ export default class MetamaskController extends EventEmitter {
     } ${hdPathDescription || ''}`.trim();
   }
 
+  /**
+   * Set account name that checks for identical label
+   *
+   * @param account
+   * @param index
+   * @param hwDeviceName
+   * @param hdPathDescription
+   */
+  setAccountName(account, index, hwDeviceName, hdPathDescription) {
+    const label = this.getAccountLabel(hwDeviceName, index, hdPathDescription);
+    try {
+      this.accountsController.setAccountName(account.id, label);
+    } catch {
+      const newIndex = index + 1;
+      this.setAccountName(account, newIndex, hwDeviceName, hdPathDescription);
```

Comment:
Ideally we'd avoid recursion so that we don't get "Maximum call stack size exceeded" errors for users with a lot of accounts. Have you considered searching for the next available default label instead?

This implementation is also problematic because we're not looking at which type of error is thrown; this would silence errors that are different than the expected one.